x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/RollupShardIndexer.java

```diff
@@ -423,13 +423,15 @@ private void collectLabel(final String field, final Object value) {
             labelFieldProducers.get(field).collect(value);
         }
 
-        private void collectMetric(final String field, final Object value) {
-            if (value instanceof Number number) {
-                metricFieldProducers.get(field).collect(number);
-            } else {
-                throw new IllegalArgumentException(
-                    "Expected numeric value for field '" + field + "' but got non numeric value: '" + value + "'"
-                );
+        private void collectMetric(final String field, final Object[] values) {
+            for (var value : values) {
+                if (value instanceof Number number) {
```

Comment:
I wonder if you could add one `instanceof` check outside the `for` loop would work.
Something like this: `if (values instanceof Number[])`

On the other hand we should totally remove the `instanceof` checks.

Reply:
Yeah, I think since we want to get the `instanceof` checks out entirely it's fine to do it either way.